vignettes/setup.Rmd

```diff
@@ -0,0 +1,206 @@
+
+```shell
+## SQL Server ODBC Drivers (Free TDS)
+apt-get install tdsodbc
+
+## PostgreSQL ODBC ODBC Drivers
+apt-get install odbc-postgresql
+
+## MySQL ODBC Drivers
+apt-get install libmyodbc
+
+## SQLite ODBC Drivers
+apt-get install libsqliteodbc
+```
+
+## Connecting to a Database
+
+Databases can be connected by specifying a connection string directly, or with DSN configuration files.
+
+### Connection Strings
+
+Specify a connection string as named arguments directly in the `dbConnect()` method.
+
+```r
+library(DBI)
+con <- 
+  dbConnect(
+    odbc::odbc(),
+    driver = "PostgreSQL Driver",
+    database = "test_db",
+    uid = "postgres",
+    pwd = "password",
+    host = "localhost",
+    port = 5432
+  )
+```
+
+Alternatively you can pass a complete connection string as the `.connection_string` argument. [The Connection Strings Reference](https://www.connectionstrings.com) is a useful resource that has example connection strings for a large variety of databases.
+
+```r
+library(DBI)
+con <- 
+  dbConnect(
+    odbc::odbc(),
+    .connection_string = "Driver={PostgreSQL Driver};Uid=postgres;Pwd=password;Host=localhost;Port=5432;Database=test_db;"
+  )
+```
+
+### DSN Configuration files
+
+ODBC configuration files are another option to specify connection parameters and allow one to use a Data Source Name (DSN) to make it easier to connect to a database.
+
+```r
+con <- dbConnect(odbc::odbc(), "PostgreSQL")
+```
+
+#### Windows
+
+The [ODBC Data Source Administrator](https://learn.microsoft.com/en-us/sql/odbc/admin/odbc-data-source-administrator?view=sql-server-ver15) application is used to manage ODBC data sources on Windows.
+
+#### MacOS / Linux
+
+On MacOS and Linux there are two separate text files that need to be edited. UnixODBC includes a command line executable `odbcinst` which can be used to query and modify the DSN files. However these are plain text files you can also edit by hand if desired.
+
+There are two different files used to setup the DSN information.
+
+- `odbcinst.ini` - which defines driver options
+- `odbc.ini` - which defines connection options
+
+The DSN configuration files can be defined globally for all users of the
+system, often at `/etc/odbc.ini`, `/opt/local/etc/odbc.ini` or `/opt/homebrew/etc/odbc.ini`, the exact location depends on what option was used when compiling unixODBC. `odbcinst -j` can be used to find the exact location. Alternatively the `ODBCSYSINI` environment variable can be used to specify the location of the configuration files. Ex. `ODBCSYSINI=~/ODBC`
+
+A local DSN file can also be used with the files `~/.odbc.ini` and `~/.odbcinst.ini`.
+
+##### MacOS aarch64
+
+If ODBC has trouble locating your system data source names, you may need to override the default location where ODBC looks for your configuration files. You can use either of the options below to specify the location of your DSN configuration files.
+
+###### Option 1: Save Setting to `~/.Renviron`
```

Comment:
Aren't these options the same?

Reply:
I can't tell the difference either. [df42053](https://github.com/r-dbi/odbc/pull/652/commits/df420535e9ca6de5190103597395fabb49ecc251).